rust_src/src/buffers.rs

```diff
@@ -129,3 +131,31 @@ pub fn buffer_file_name(buffer: LispObject) -> LispObject {
 
     LispObject::from_raw(buf.filename)
 }
+
+
+#[no_mangle]
+pub extern "C" fn validate_region(b: *mut Lisp_Object, e: *mut Lisp_Object) {
+    let mut start = LispObject::from_raw(unsafe { *b });
+    let mut stop = LispObject::from_raw(unsafe { *e });
+
+    let mut beg = start.as_fixnum_coerce_marker_or_error();
+    let mut end = stop.as_fixnum_coerce_marker_or_error();
+
+    if beg > end {
+        mem::swap(&mut beg, &mut end);
+    }
+
+    unsafe {
+        *b = start.to_raw();
```

Comment:
I fixed it in the latest commit; I swap start and end as well. It seems to work.